File: src/Entity/FightResult.php
<?php
namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\FightResultRepository;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: FightResultRepository::class)]
#[ORM\Table(name: 'fight_results')]
class FightResult
{
    public const METHOD_KO = 'KO';
    public const METHOD_DQ = 'DQ';
    public const METHOD_DRAW = 'DRAW';
    public const METHOD_DECISION = 'DECISION';

    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'resultId')]
    private ?int $resultId = null;

    #[ORM\ManyToOne(targetEntity: Event::class)]
    #[ORM\JoinColumn(name: 'eventId', referencedColumnName: 'eventId', nullable: false)]
    private ?Event $event = null;

    #[ORM\Column(name: 'fightNumber')]
    private int $fightNumber = 1;

    #[ORM\ManyToOne(targetEntity: Fighter::class)]
    #[ORM\JoinColumn(name: 'fighter1Id', referencedColumnName: 'fighterId', nullable: false)]
    private ?Fighter $fighter1 = null;

    #[ORM\ManyToOne(targetEntity: Fighter::class)]
    #[ORM\JoinColumn(name: 'fighter2Id', referencedColumnName: 'fighterId', nullable: false)]
    private ?Fighter $fighter2 = null;

    #[ORM\ManyToOne(targetEntity: Fighter::class)]
    #[ORM\JoinColumn(name: 'winnerId', referencedColumnName: 'fighterId', nullable: true)]
    private ?Fighter $winner = null;

    #[ORM\Column(name: 'methodOfVictory', length: 50, nullable: true)]
    #[Assert\Choice(choices: [self::METHOD_KO, self::METHOD_DQ, self::METHOD_DRAW, self::METHOD_DECISION])]
    private ?string $methodOfVictory = null;

    #[ORM\Column(name: 'decision_type', length: 10, nullable: true)]
    #[Assert\Choice(choices: ['UD', 'SD', 'MD'])]
    private ?string $decisionType = null;

    #[ORM\Column(name: 'knockdown_round', type: 'integer', nullable: true)]
    #[Assert\Positive]
    #[Assert\LessThanOrEqual(12)]
    private ?int $knockdownRound = null;

    #[ORM\Column(name: 'roundNumber', nullable: true)]
    #[Assert\Positive]
    #[Assert\LessThanOrEqual(12)]
    private ?int $roundNumber = null;

    #[ORM\Column(name: 'scheduled_rounds', type: 'integer', options: ['default' => 12])]
    #[Assert\Choice(choices: [4, 6, 8, 10, 12])]
    private int $scheduledRounds = 12;

    #[ORM\Column(name: 'is_belt_fight', type: 'boolean', options: ['default' => false])]
    private bool $isBeltFight = false;

    #[ORM\Column(name: 'belt_organization', length: 20, nullable: true)]
    #[Assert\Choice(choices: ['WBC', 'WBA', 'IBF', 'WBO', 'UNDISPUTED'])]
    private ?string $beltOrganization = null;

    #[ORM\Column(name: 'fighter1_odds', type: 'float', nullable: true)]
    #[Assert\Positive]
    private ?float $fighter1Odds = null;

    #[ORM\Column(name: 'fighter2_odds', type: 'float', nullable: true)]
    #[Assert\Positive]
    private ?float $fighter2Odds = null;

    #[ORM\Column(name: 'fightDate', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $fightDate = null;

    #[ORM\Column(length: 20, options: ['default' => 'SCHEDULED'])]
    private string $status = 'SCHEDULED';

    public function getResultId(): ?int { return $this->resultId; }
    public function getEvent(): ?Event { return $this->event; }
    public function setEvent(Event $v): self { $this->event = $v; return $this; }
    public function getFightNumber(): int { return $this->fightNumber; }
    public function setFightNumber(int $v): self { $this->fightNumber = $v; return $this; }
    public function getFighter1(): ?Fighter { return $this->fighter1; }
    public function setFighter1(Fighter $v): self { $this->fighter1 = $v; return $this; }
    public function getFighter2(): ?Fighter { return $this->fighter2; }
    public function setFighter2(Fighter $v): self { $this->fighter2 = $v; return $this; }
    public function getWinner(): ?Fighter { return $this->winner; }
    public function setWinner(?Fighter $v): self { $this->winner = $v; return $this; }
    public function getMethodOfVictory(): ?string { return $this->methodOfVictory; }
    public function setMethodOfVictory(?string $v): self { $this->methodOfVictory = $v; return $this; }
    
    public function getDecisionType(): ?string { return $this->decisionType; }
    public function setDecisionType(?string $v): self { $this->decisionType = $v; return $this; }

    public function getKnockdownRound(): ?int { return $this->knockdownRound; }
    public function setKnockdownRound(?int $v): self { $this->knockdownRound = $v; return $this; }

    public function getRoundNumber(): ?int { return $this->roundNumber; }
    public function setRoundNumber(?int $v): self { $this->roundNumber = $v; return $this; }
    
    public function getScheduledRounds(): int { return $this->scheduledRounds; }
    public function setScheduledRounds(int $v): self { $this->scheduledRounds = $v; return $this; }

    public function isBeltFight(): bool { return $this->isBeltFight; }
    public function setIsBeltFight(bool $v): self { $this->isBeltFight = $v; return $this; }

    public function getBeltOrganization(): ?string { return $this->beltOrganization; }
    public function setBeltOrganization(?string $v): self { $this->beltOrganization = $v; return $this; }

    public function getFighter1Odds(): ?float { return $this->fighter1Odds; }
    public function setFighter1Odds(?float $v): self { $this->fighter1Odds = $v; return $this; }

    public function getFighter2Odds(): ?float { return $this->fighter2Odds; }
    public function setFighter2Odds(?float $v): self { $this->fighter2Odds = $v; return $this; }
    public function getFightDate(): ?\DateTimeInterface { return $this->fightDate; }
    public function setFightDate(?\DateTimeInterface $v): self { $this->fightDate = $v; return $this; }
    public function getStatus(): string { return $this->status; }
    public function setStatus(string $v): self { $this->status = $v; return $this; }

    public function getLoser(): ?Fighter
    {
        if ($this->winner === null || $this->status !== 'COMPLETED') return null;
        return ($this->winner === $this->fighter1) ? $this->fighter2 : $this->fighter1;
    }

    public function getMethodBonus(): int
    {
        if ($this->methodOfVictory === null) return 0;
        return match (strtoupper($this->methodOfVictory)) {
            self::METHOD_KO => 10,
            self::METHOD_DECISION => 5,
            self::METHOD_DQ => 2,
            self::METHOD_DRAW => 1,
            default => 0,
        };
    }

    public function isDraw(): bool
    {
        return $this->winner === null && $this->status === 'COMPLETED' && $this->methodOfVictory === self::METHOD_DRAW;
    }
}

File: src/main/java/tn/smartfight/model/FightResult.java
package tn.smartfight.model;

import java.time.LocalDateTime;

public class FightResult {
    private int resultId;
    private int fightNumber;
    private String methodOfVictory;
    private String decisionType;
    private int knockdownRound;
    private int roundNumber;
    private int scheduledRounds = 12;
    private boolean beltFight;
    private String beltOrganization;
    private double fighter1Odds;
    private double fighter2Odds;
    private LocalDateTime fightDate;
    private String status = "SCHEDULED";
    private int eventId;
    private int fighter1Id;
    private int fighter2Id;
    private Integer winnerId;
    private String insideTheNumbers;
    private String highlightVideoUrl;
    private String videoPath;
    private LocalDateTime updatedAt;

    // display-only (joined from related tables)
    private String eventName;
    private String fighter1Name;
    private String fighter2Name;
    private String winnerName;

    public FightResult() {}

    public int getResultId() { return resultId; }
    public void setResultId(int resultId) { this.resultId = resultId; }

    public int getFightNumber() { return fightNumber; }
    public void setFightNumber(int fightNumber) { this.fightNumber = fightNumber; }

    public String getMethodOfVictory() { return methodOfVictory; }
    public void setMethodOfVictory(String methodOfVictory) { this.methodOfVictory = methodOfVictory; }

    public String getDecisionType() { return decisionType; }
    public void setDecisionType(String decisionType) { this.decisionType = decisionType; }

    public int getKnockdownRound() { return knockdownRound; }
    public void setKnockdownRound(int knockdownRound) { this.knockdownRound = knockdownRound; }

    public int getRoundNumber() { return roundNumber; }
    public void setRoundNumber(int roundNumber) { this.roundNumber = roundNumber; }

    public int getScheduledRounds() { return scheduledRounds; }
    public void setScheduledRounds(int scheduledRounds) { this.scheduledRounds = scheduledRounds; }

    public boolean isBeltFight() { return beltFight; }
    public void setBeltFight(boolean beltFight) { this.beltFight = beltFight; }

    public String getBeltOrganization() { return beltOrganization; }
    public void setBeltOrganization(String beltOrganization) { this.beltOrganization = beltOrganization; }

    public double getFighter1Odds() { return fighter1Odds; }
    public void setFighter1Odds(double fighter1Odds) { this.fighter1Odds = fighter1Odds; }

    public double getFighter2Odds() { return fighter2Odds; }
    public void setFighter2Odds(double fighter2Odds) { this.fighter2Odds = fighter2Odds; }

    public LocalDateTime getFightDate() { return fightDate; }
    public void setFightDate(LocalDateTime fightDate) { this.fightDate = fightDate; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public int getEventId() { return eventId; }
    public void setEventId(int eventId) { this.eventId = eventId; }

    public int getFighter1Id() { return fighter1Id; }
    public void setFighter1Id(int fighter1Id) { this.fighter1Id = fighter1Id; }

    public int getFighter2Id() { return fighter2Id; }
    public void setFighter2Id(int fighter2Id) { this.fighter2Id = fighter2Id; }

    public Integer getWinnerId() { return winnerId; }
    public void setWinnerId(Integer winnerId) { this.winnerId = winnerId; }

    public String getInsideTheNumbers() { return insideTheNumbers; }
    public void setInsideTheNumbers(String insideTheNumbers) { this.insideTheNumbers = insideTheNumbers; }

    public String getHighlightVideoUrl() { return highlightVideoUrl; }
    public void setHighlightVideoUrl(String highlightVideoUrl) { this.highlightVideoUrl = highlightVideoUrl; }

    public String getVideoPath() { return videoPath; }
    public void setVideoPath(String videoPath) { this.videoPath = videoPath; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    public String getEventName() { return eventName; }
    public void setEventName(String eventName) { this.eventName = eventName; }

    public String getFighter1Name() { return fighter1Name; }
    public void setFighter1Name(String fighter1Name) { this.fighter1Name = fighter1Name; }

    public String getFighter2Name() { return fighter2Name; }
    public void setFighter2Name(String fighter2Name) { this.fighter2Name = fighter2Name; }

    public String getWinnerName() { return winnerName; }
    public void setWinnerName(String winnerName) { this.winnerName = winnerName; }
}
